src/barChartSettingsModel.ts

```diff
@@ -0,0 +1,181 @@
+import powerbiVisualsApi from "powerbi-visuals-api";
+import { formattingSettings } from "powerbi-visuals-utils-formattingmodel";
+import { BarChartDataPoint } from "./barChart";
+
+import Card = formattingSettings.SimpleCard;
+import Model = formattingSettings.Model;
+
+class EnableAxisCardSettings extends Card {
+    show = new formattingSettings.ToggleSwitch({
+        name: "show",
+        displayName: undefined,
+        value: false,
+    });
+
+    fill = new formattingSettings.ColorPicker({
+        name: "fill",
+        displayName: "Color",
+        value: { value: "#000000" }
+    });
+    topLevelSlice = this.show;
+    name: string = "enableAxis";
+    displayName: string = "Enable Axis";
+    slices = [this.show, this.fill];
+}
+
+
+class ColorSelectorCardSettings extends Card {
+    name: string = "colorSelector";
+    displayName: string = "Data Colors";
+    slices = [];
+}
+
+class GeneralViewCardSettings extends Card {
+    opacity = new formattingSettings.NumUpDown({
+        name: "opacity",
+        displayName: "Bars Opacity",
+        value: 100,
+        options: {
+            minValue: {
+                type: powerbiVisualsApi.visuals.ValidatorType.Min,
+                value: 0,
+            },
+            maxValue: {
+                type: powerbiVisualsApi.visuals.ValidatorType.Max,
+                value: 100,
+            }
+        }
+    });
+
+    showHelpLink = new formattingSettings.ToggleSwitch({
+        name: "showHelpLink",
+        displayName: "Show Help Button",
+        value: false
+    });
+
+    name: string = "generalView";
+    displayName: string = "General View";
+    helpLinkColor: string = "#80B0E0"
+    slices = [this.opacity, this.showHelpLink];
+}
+
+class AverageLineCardSettings extends Card {
+    show = new formattingSettings.ToggleSwitch({
+        name: "show",
+        displayName: undefined,
+        value: false,
+    });
+
+    fill = new formattingSettings.ColorPicker({
+        name: "fill",
+        displayName: "Color",
+        value: { value: "#888888" },
+    });
+
+    showDataLabel = new formattingSettings.ToggleSwitch({
+        name: "showDataLabel",
+        displayName: "Data Label",
+        value: false
+    });
+
+    topLevelSlice = this.show;
+    name: string = "averageLine";
+    displayName: string = "Average Line";
+    analyticsPane: boolean = true;
+    slices = [this.show, this.fill, this.showDataLabel];
```

Comment:
TopLevelSlice shouldn't be in slices as it'll be duplicated

Reply:
removed from the slices array